### PR TITLE
fix(diff): expand untracked directories for preview

### PR DESCRIPTION
**Pull request by chuxubank**
Untracked files nested inside new directories can now be opened in the native DIFF preview.

## What

- ask Git to enumerate every untracked file instead of collapsing new directories
- add a regression test covering a text file inside a nested untracked directory
- preserve the existing bounded status scan and regular-file safety checks

## Why

`git status --porcelain=v2` collapses an untracked directory to a single `directory/` entry by default. Luvus then attempted to load that entry as a file, so DIFF preview rejected it as a non-regular file.

## Verification

- [x] `cargo test --locked`
- [x] `cargo clippy --all-targets -- -D warnings`
- [x] `cargo fmt --all --check`
- [ ] Manual testing, if applicable

## Notes

- Empty directories remain absent because Git does not track them.
- Untracked symlink targets remain intentionally unrendered.

Before:
file-1da5873e2a019e344a46550dbd33d59f

After:
file-92d2d8ad8fd34371e87aac3deb85a73a

<h2><a href="https://app.greptile.com/api/retrigger?id=66601159"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/RetriggerDark.svg?v=2"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/Retrigger.svg?v=2">Retrigger</picture></a></h2>

Safe to merge.

<details><summary>Summary</summary>

The update preserves tracked changes when bounded untracked-file discovery reaches its output limit, while keeping only complete untracked path records. It also correctly retains a final pathname whose NUL terminator immediately follows the configured content limit.
</details>

<sub>Reviews (3) · Last reviewed commit: ["fix(diff): retain boundary untracked rec..."](https://github.com/rizriyz/luvus/commit/cf40933dbc9d8c2234a988cd27591a29fde93284)</sub>

### CODE REVIEW

File: src/diff/git.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ pub fn scan(root: &Path, generation: u64) -> Result<DiffSnapshot, String> {
         &[
             OsString::from("status"),
             OsString::from("--porcelain=v2"),
+            OsString::from("--untracked-files=all"),
```
[the remark this answers is not in the document]
Comment (chuxubank) [reply src/diff/git.rs]: Fixed in 7bb5dfd. Tracked status now runs separately with `--untracked-files=no`, while untracked paths are enumerated through a bounded `git ls-files --others --exclude-standard -z` read. If that listing exceeds the cap, Luvus keeps tracked changes, drops any partial trailing pathname, retains only complete previewable entries, and marks the snapshot as having omitted files. Added `oversized_untracked_listing_preserves_tracked_changes` to exercise the real scanner with a deliberately small untracked-output cap. Verified with focused DIFF tests, strict Clippy, formatting, the locked release build, and the full test binary (1959 passed, 5 ignored).

File: src/diff/git.rs [hunk 2]
```diff
@@ -24,20 +32,50 @@ pub fn scan(root: &Path, generation: u64) -> Result<DiffSnapshot, String> {
...
+    )?;
+    if untracked_truncated {
+        // The bounded read may end halfway through a pathname. Publish only
+        // complete NUL-delimited records so every listed entry remains safe to
+        // open; `omitted_files` below advertises that the list is incomplete.
+        let complete = untracked_raw
+            .iter()
+            .rposition(|byte| *byte == 0)
+            .map_or(0, |end| end + 1);
+        untracked_raw.truncate(complete);
```
[the remark this answers is not in the document]
Comment (chuxubank) [reply src/diff/git.rs:63]: Fixed in cf40933. NUL-delimited Git reads now allow one structural delimiter byte beyond the pathname-content cap while retaining a further lookahead byte for truncation detection. A pathname that exactly fills the cap and is followed by its terminating NUL is therefore kept without reporting an omission; genuinely partial or additional records remain bounded and use the existing complete-record trimming. Added `final_untracked_nul_just_past_cap_keeps_the_complete_path` for the exact boundary case. Verified with all 13 DIFF Git tests, strict Clippy, formatting, locked release build, and the full test binary (1960 passed, 5 ignored).

**Review by RizRiyz — APPROVED** [review]

**Comment by RizRiyz** [comment]
Approved! thank you @chuxubank
